ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java

```diff
@@ -597,14 +614,16 @@ public Map<String, CloudBlobMetadata> getBlobMetadata(List<BlobId> blobIds) thro
       AzureBlobLayoutStrategy.BlobLayout blobLayout = this.azureBlobLayoutStrategy.getDataBlobLayout(blobId);
       try {
         BlobProperties blobProperties = getBlobProperties(blobLayout);
-        cloudBlobMetadataMap.put(blobId.getID(), CloudBlobMetadata.fromMap(blobProperties.getMetadata()));
-      } catch (Throwable t) {
-        if (t instanceof CloudStorageException
-            && ((CloudStorageException) t).getStatusCode() == CloudBlobStore.STATUS_NOT_FOUND) {
-          // We should never be here because replication logic checks if a blob exists or not before undeleting it.
-          String dbg = String.format("Blob %s absent in Azure blob storage", blobLayout);
-          logger.error(dbg);
+        // We are trying to find blobs present on servers but absent in cloud. Many PUTs fall into this bucket.
+        // Do not print any errors as this is expected when the servers are ahead of cloud and cloud is a bit stale.
+        // blobProperties == null when a blob is absent in cloud.
+        if (blobProperties != null) {
```

Comment:
Just let getBlobProperties continues to throw NOT_FOUND as most of our code does.
Here if the exception is not_FOUND, swallow it or print with trace level.

Reply:
done